core: set gasLimit in '--dev' mode to be 9000000

madjarevicn opened the pull request:
In order to make testing in --dev mode more used, and support more cases, I changed the `gasLimit` in `--dev` mode to 9 milion. Mainnet currenly supports almost 10m, which is much closer to it.

karalabe commented:
Mainnet currently is at 12M, might as well bump it all the way there.

karalabe commented:
Btw, I also saw that in dev mode we don't have 1 wei allocations for newer precompiler. Might be worthwhile to fix those too at the same time.

madjarevicn commented:
Yeah, 12M is even better.
I'll submit new PR with that, and also I can add 1 wei allocations for
newer precompiles.
Let me know if OK, and I'll get to it today.

Best regards,
Nikola

On Wed, Jul 15, 2020 at 10:11 AM Péter Szilágyi <notifications@github.com>
wrote:

> Btw, I also saw that in dev mode we don't have 1 wei allocations for newer
> precompiles. Might be worthwhile to fix those too at the same time.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ethereum/go-ethereum/pull/21323#issuecomment-658617336>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHFK4LB7FH7CYJX2PHGH6EDR3VQEVANCNFSM4OYANR7A>
> .
>


-- 
~*~

*Nikola Madjarevic*

Director of Blockchain R&D

M: +(381) 62-1071624

TG: @nikolamadjarevic



2key.network
Join Us On Telegram <http://bit.ly/2keyTG>

karalabe commented:
sgtm

madjarevicn commented:
Hey Folks. 

I have added also 1 wei allocation for newer precompiles, and updated gasLimit to 11.5 milion, since regarding June 21st 2020, blocks are closely 12m, so 0.5m less would assure that everything working on `--dev` should work on mainnet as well.

dahu33 commented:
Not sure if this was intended but this change was a breaking change: if you had an existing chain in '--dev' mode, geth v1.9.17 will refuse to start with:
```
Fatal: Error starting protocol stack: database contains incompatible genesis (have f60d6c04a6b6264f1c89845b697293e2c242bee3d8463968848b16198a7c2807, new 1bf2ffd2d8981fda0a342215ef5988889fe76cf613a6a744fd663da7e760576b)
```

karalabe commented:
@dahu33 Hmm, that's odd. Geth should load the genesis from disk. Otherwise the random signer funding wouldn't work across restarts either. Lemme take a closer look.

karalabe commented:
@dahu33 Could you try if https://github.com/ethereum/go-ethereum/pull/21352 fixes it for you?

dahu33 commented:
> @dahu33 Could you try if #21352 fixes it for you?

Thanks for the quick fix, #21352 fixes it.
